stdlib/public/core/UnsafePointer.swift.gyb

```diff
@@ -157,8 +157,9 @@ public struct ${Self}<Pointee>
 
   /// Accesses the `Pointee` instance referenced by `self`.
   ///
-  /// - Precondition: the pointee has been initialized with an instance of
-  ///   type `Pointee`.
+  /// - Precondition: the pointee has been initialized with an instance of type
+  ///   `Pointee` unless `pointee` is the left side of an assignment and
+  ///   `Pointee` is a trivial type.
```

Comment:
This could be read to mean that if `pointee` is the LHS and `Pointee` is a trivial type then the pointee shouldn't be initialized.

I'd suggest something along the lines of

```swift
/// - Precondition: Either the pointee has been initialized with an instance of type
///   `Pointee`, or `pointee` is being assigned to and `Pointee` is a trivial type.
```